tests/lib/structures/Josh.test.ts

```diff
@@ -11,7 +13,121 @@ describe('Josh', () => {
     });
   });
 
-  describe('methods runs', () => {
+  describe('can be instantiated class', () => {
+    test('GIVEN class Josh THEN returns Josh', () => {
+      const josh = new Josh({ name: 'test:name' });
+      expect(josh).toBeInstanceOf(Josh);
+    });
+
+    test('GIVEN class Josh w/o name THEN throws error', () => {
+      expect(() => new Josh({ name: undefined })).toThrowError('The "name" option is required to initiate a Josh instance');
+    });
+
+    test('GIVEN class Josh w/ autoEnsure THEN returns middleware size 1', () => {
+      const josh = new Josh({ name: 'test:name', autoEnsure: { defaultValue: { foo: 'bar' } } });
+      expect(josh.middlewares.size).toBe(1);
+    });
+
+    test('GIVEN class Josh w/ empty middlewares THEN returns instance', () => {
+      const josh = new Josh({ name: 'test:name', middlewares: [] });
+      expect(josh.middlewares.size).toBe(0);
+    });
+
+    test('GIVEN class Josh w/ middlewares THEN returns instance', () => {
+      const josh = new Josh<{ test: boolean }>({
+        name: 'test:name',
+        // For nova, this seems to be the only way to make sure the middleware is allowed
+        middlewares: [new AutoEnsureMiddleware({ defaultValue: { test: false } }) as unknown as Middleware<{ test: boolean }>]
+      });
+
+      expect(josh.middlewares.size).toBe(1);
+    });
+
+    test('GIVEN provider with invalid instance THEN emits warning', () => {
+      const spy = jest.spyOn(process, 'emitWarning').mockImplementation();
+      // @ts-expect-error - this is a test
+      const josh = new Josh({ name: 'test:instanceof', provider: new Map() });
```

Comment:
what are the errors that come from this?

Reply:
Map isn't a JoshProvider

Reply:
we have the `@joshdb/map` dep so we could use that?

Reply:
oh nvm